### PR TITLE
Set X-Presto-User header when reading nextUri

### PR DESCRIPTION
**Pull request by xerial**
Even when reading a query result, we would like to check who is accessing the nextUri. This PR sets X-Presto-User header while reading next statement result. For example, some service access token can be set here to authenticate each request.

This would be an easier alternative for #4928.

### CODE REVIEW

**Comment by electrum** [comment]
I assume you want this to prevent spoofing? As an alternative, what if we made the nextUri non spoofable? For example, by including an HMAC of the URI info plus a random secret in the coordinator's memory?

**Comment by xerial** [comment]
HMAC is good for preventing spoofing, but we also need to label each HTTP request (with X-Presto-User or X-Presto-Connection-Property #4928) so that we can dispatch the incoming request to an appropriate presto coordinator or dropping unauthorized requests.

**Comment by xerial** [comment]
And also we are considering rewriting nextUri so that we can create a proxy server (e.g., presto-gateway.xxx.com) for accessing multiple Presto clusters, which return private IP addresses. So encrypting nextUri in anyway is not preferable.

**Comment by xerial** [comment]
@electrum FYI, Teradata's Presto ODBC driver also set X-Presto-User header while reading /v1/statement/(query_id)/N

**Comment by xerial** [comment]
@electrum We just need to make sure that X-Presto-User header is set when reading /v1/statement/(query_id)/N. For security, we still need to use HTTPS connection or something else.

File: presto-client/src/main/java/com/facebook/presto/client/StatementClient.java [hunk 1]
```diff
@@ -101,6 +103,7 @@ public StatementClient(HttpClient httpClient, JsonCodec<QueryResults> queryResul
         this.timeZoneId = session.getTimeZoneId();
         this.query = query;
         this.requestTimeoutNanos = session.getClientRequestTimeout().roundTo(NANOSECONDS);
+        this.user = Optional.ofNullable(session.getUser());
```
Comment (martint) [inline presto-client/src/main/java/com/facebook/presto/client/StatementClient.java]: The user cannot be missing, so no need to make this an optional.

File: presto-client/src/main/java/com/facebook/presto/client/StatementClient.java [hunk 2]
```diff
@@ -132,7 +131,6 @@ private static Request buildQueryRequest(ClientSession session, String query)
         }
         builder.setHeader(PrestoHeaders.PRESTO_TIME_ZONE, session.getTimeZoneId());
         builder.setHeader(PrestoHeaders.PRESTO_LANGUAGE, session.getLocale().toLanguageTag());
-        builder.setHeader(USER_AGENT, USER_AGENT_VALUE);
```
Comment (martint) [inline presto-client/src/main/java/com/facebook/presto/client/StatementClient.java]: why did you move this out? The user agent should always be set.

File: presto-client/src/main/java/com/facebook/presto/client/StatementClient.java (hunk 1 above)
Comment (xerial) [inline presto-client/src/main/java/com/facebook/presto/client/StatementClient.java]: OK. I'll remove Optional.

File: presto-client/src/main/java/com/facebook/presto/client/StatementClient.java (hunk 2 above)
Comment (xerial) [inline presto-client/src/main/java/com/facebook/presto/client/StatementClient.java]: I moved this code inside `prepareRequest` https://github.com/prestodb/presto/pull/4929/files#diff-9c9280d592f8582c73302c8114b44a29R222 so as not to forget to set USER_AGENT.

**Comment by xerial** [comment]
@martint Removed the use of Optional as you commented.

**Comment by martint** [comment]
I'll merge this after the current release goes out (in progress)

**Comment by xerial** [comment]
Thanks!

**Comment by martint** [comment]
Merged, thanks!
